### PR TITLE
Feature/fredeluga spoofer

### PR DESCRIPTION
**Pull request by marija-p**
- General clean-up of GPS spoofer
- Resolved parameter namespace issues
- Added parameter to publish disturbed pose or not (true/false) to be specified in launch file

@enricgalceran @raghavkhanna

### CODE REVIEW

File: src/gps_spoofer.py [hunk 1]
```diff
@@ -100,8 +99,8 @@ def callback_odometry(self, data):
         self.pwc.pose.pose.position.z = data.pose.pose.position.z
         self.point.point.z = data.pose.pose.position.z
     elif (self.altitude_input == 'external'):
-        self.pwc.pose.pose.position.z = self.altitude.data
-        self.point.point.z = self.altitude.data
+        self.pwc.pose.pose.position.z = self.latest_altitude_message.data
+        self.point.point.z = self.latest_altitude_message.data
     else:
         ROS_WARN("Unknown altitude input parameter")
```
Comment (enricgalceran) [inline src/gps_spoofer.py]: @marija-p Kill the node here when this happens.

File: src/gps_spoofer.py [hunk 2]
```diff
@@ -12,6 +12,16 @@
 
 class GPSSpoofer:
   def __init__(self):
+    # Parameters
+    self.altitude_input = rospy.get_param('~altitude_input', 'vicon')
+    self.pose_publisher = rospy.get_param('~publish_pose', True)
```
Comment (enricgalceran) [inline src/gps_spoofer.py]: Why is this named 'publisher'? Variable name should match parameter name.

**Comment by marija-p** [comment]
@enricgalceran  I've changed both now, but I'm not sure if the method of shutting down the node is correct (http://wiki.ros.org/rospy/Overview/Initialization%20and%20Shutdown)

**Comment by enricgalceran** [comment]
You should call sys.exit() afterwards too.

**Comment by marija-p** [comment]
Thanks! :+1:

**Comment by marija-p** [comment]
All ok now? @enricgalceran

**Comment by enricgalceran** [comment]
LGTM.

**Comment by marija-p** [comment]
Merging

**Pull request merged**